--- busybox-1.6.0/init/init.c	2007-06-01 13:48:40.000000000 +0200
+++ src/init/init.c	2007-06-07 18:52:12.000000000 +0200
@@ -938,7 +938,7 @@
 	if (argc > 1) setenv("RUNLEVEL", argv[1], 1);
 
 	/* Hello world */
-	message(MAYBE_CONSOLE | L_LOG, "init started: %s", bb_msg_full_version);
+	message(MAYBE_CONSOLE, "init started: %s", bb_msg_full_version);
 
 	/* Make sure there is enough memory to do something useful. */
 	if (ENABLE_SWAPONOFF) {
